packages/SwingSet/src/liveslots/liveslots.js

```diff
@@ -905,6 +923,14 @@ function build(
     const maybeNewVPIDs = new Set(serMethargs.slots);
     const resolutions = resolutionCollector().forSlots(serMethargs.slots);
     if (resolutions.length > 0) {
+      try {
```

Comment:
We should do the size check *before* calling `syscall.resolve`, otherwise we're making it easy for userspace to trigger the netstring size limit (which will be a kernel panic, rather than an in-consensus fatal vat error).

Also, we can simplify things a little bit because it's safe to build up `maybeNewVPIDs` before the call to `syscall.resolve` (it's just a local Set).

So move the `resolutions.forEach(.. maybeNewVPIDs.add)` to the line before `syscall.resolve()`, and change it to do both the `add` and the `try(assert..) catch {exit..}` clause inside the loop. For tidyness, we can move the second loop (the `maybeNewVPIDs.delete` one) to before the `syscall.resolve` too. Neither has any externally-visible side-effects by this point in the function.

Reply:
I'm confused by this comment.  We _do_ do the size check before calling `syscall.resolve`.  Are you possibly mixing up the two calls to `resolutions.forEach` with each other when reading the code?

Reply:
Checking the multiple capdatas of a resolve collectively undoes the benefit of the proposed refactoring, since we have to iterate the resolution collection to extract the capdatas as an array, so I'm skipping this for now.  You might want to take a look when you review the revisions to see if you spot a cleaner way that I didn't spot.